tgg/seqr_scripts/pull_seqr_pedigrees.py

```diff
@@ -2,46 +2,74 @@
 import logging
 
 import requests
+from typing import Set
 
 logging.basicConfig(
-    format="%(asctime)s): %(message)s", datefmt="%m/%d/%Y %I:%M:%S %p",
+    format="%(asctime)s): %(message)s",
+    datefmt="%m/%d/%Y %I:%M:%S %p",
 )
 logger = logging.getLogger()
 logger.setLevel(logging.INFO)
 
 SEQR_URL = "https://seqr.broadinstitute.org"
+GET_FAMILIES_URL = "get_families"
+GET_INDIVIDUALS_URL = "get_individuals"
 
-def pull_project_peds(session_id: str, projects: set):
+
+def pull_project_peds(session_id: str, projects: Set[str]):
     """
-    Uses the seqr API to retrieve a list of projects' pedigrees. Writes a single pedigree,
-    'seqr_pedigrees.txt', using the retrieved pedigrees. Failing projects are written
-    to 'projects_not_pulled.txt'
+    Use the seqr API to retrieve a list of projects' pedigrees.
+
+    Writes a single pedigree for all projects ('seqr_pedigrees.txt'), using the retrieved pedigrees.
+    Failing projects are written to 'projects_not_pulled.txt'.
+
     :param session_id: sessionid cookie used for seqr login
-    :param projects:  set of seqr project GUIDs
+    :param projects:  Set of seqr project GUIDs
     """
     with requests.Session() as s:
         # Initialize session with authenticated cookie
-        s.cookies.set_cookie(requests.cookies.create_cookie('sessionid', session_id))
-
-        with open("seqr_pedigrees.txt", "w") as final_ped, open("projects_not_pulled.txt", "w") as errors:
+        s.cookies.set_cookie(requests.cookies.create_cookie("sessionid", session_id))
+        with open("seqr_pedigrees.txt", "w") as final_ped, open(
+            "projects_not_pulled.txt", "w"
+        ) as errors:
             final_ped.write(
                 "Project_GUID\tFamily_ID\tIndividual_ID\tPaternal_ID\tMaternal_ID\tSex\n"
             )
             for project_guid in projects:
                 project_guid = project_guid.rstrip()
-                r = s.get(
-                    f"{SEQR_URL}/api/project/{project_guid}/details"
+                family_r = s.get(
+                    f"{SEQR_URL}/api/project/{project_guid}/{GET_FAMILIES_URL}"
                 )
-
-                if r.status_code != requests.codes["ok"]:
-                    logger.info(f"Could not find {project_guid} in seqr")
+                individual_r = s.get(
+                    f"{SEQR_URL}/api/project/{project_guid}/{GET_INDIVIDUALS_URL}"
+                )
+                if (family_r.status_code != requests.codes["ok"]) or (
+                    individual_r.status_code != requests.codes["ok"]
+                ):
```

Comment:
are these lines needed? Either scenario is already accounted for below right?

Reply:
hmmm this is a good point. you have these lines here to group these two errors together with the error message (but also print out which call failed), right? what if you change this to something like
```
if (family_r.status_code != requests.codes["ok"]) or (
    individual_r.status_code != requests.codes["ok"]
):
    if family_r.status_code != requests.codes["ok"]:
            logger.info("...")
    if individual_r.status_code != requests.codes["ok"]:
            logger.info("..")
    errors.write(f"{project_guid}\n")
    continue

fams = ...
inds = ...
...
```